src/lib/linux-api/src/posix_types.rs

```diff
@@ -15,3 +17,26 @@ pub type kernel_ulong_t = bindings::linux___kernel_ulong_t;
 pub use bindings::linux___kernel_off_t;
 #[allow(non_camel_case_types)]
 pub type kernel_off_t = linux___kernel_off_t;
+
+/// Type-safe wrapper around [`kernel_pid_t`]. Value is strictly positive.
+/// Interface inspired by `rustix::process::Pid`.
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+pub struct Pid(NonZeroI32);
+
+impl Pid {
+    pub fn from_raw(pid: kernel_pid_t) -> Option<Self> {
+        if pid < 0 {
+            None
+        } else {
+            Some(Self(NonZeroI32::new(pid).unwrap()))
+        }
+    }
+
+    pub fn as_raw(this: Option<Self>) -> kernel_pid_t {
+        this.map(|x| kernel_pid_t::from(x.0)).unwrap_or(0)
+    }
```

Comment:
nit: Maybe a little confusing since the value will not necessarily be the same after a round trip (for example `Pid::as_raw(Pid::from_raw(-1)) == 0`). I think this should at least document that 0 is returned for `None`).

Reply:
Done